src/main/java/com/google/devtools/build/lib/actions/ActionInputMap.java

```diff
@@ -111,6 +112,13 @@ public ActionInput getInput(String execPathString) {
     return index == -1 ? null : (ActionInput) keys[index];
   }
 
+  @Override
+  public boolean artifactOmitted(Artifact artifact) {
+    // This class does not implement markOmitted() method so call to `artifactOmitted()`
+    // means logic error.
+    throw new UnsupportedOperationException();
```

Comment:
wouldn't `return false` be correct here?

Reply:
Done.